hal/sdio_api.h

```diff
@@ -0,0 +1,164 @@
+
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_SDIO_API_H
+#define MBED_SDIO_API_H
+
+#include "device.h"
+#include "pinmap.h"
+
+#if DEVICE_SDIO
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * \defgroup SDIO Configuration Functions
+ * @{
+ */
+
+typedef struct {
+    uint32_t CardType;     /* Specifies the card Type                         */
+    uint32_t CardVersion;  /* Specifies the card version                      */
+    uint32_t Class;        /* Specifies the class of the card class           */
+    uint32_t RelCardAdd;   /* Specifies the Relative Card Address             */
+    uint32_t BlockNbr;     /* Specifies the Card Capacity in blocks           */
+    uint32_t BlockSize;    /* Specifies one block size in bytes               */
+    uint32_t LogBlockNbr;  /* Specifies the Card logical Capacity in blocks   */
```

Comment:
Will be good to add some comments about the difference between  BlockNbr and LogBlockNbr

Reply:
this part is a direct copy from sdio-driver.
@JojoS62 is there a difference?